testing/api/test_api.py

```diff
@@ -1136,3 +1148,230 @@ def test_create_invalid_chars(empty_devices_dir, testrun): # pylint: disable=W06
+
+    #check that "questions" has the expected fields
+    for element in profile["questions"]:
+      assert isinstance(element, dict) #check if each element is dict
+      assert "question" in element
+      assert "type" in element
+      assert "answer" in element
+
+#test for create profile if not exists
+def test_create_profile(testrun, clean_profiles_dir): # pylint: disable=W0613
+  new_profile = load_json("new_profile.json")
+  profile_name = new_profile["name"]
+
+  assert not profile_exists(profile_name), f"Profile '{profile_name}' exists"
+
+  r = requests.post(f"{API}/profiles", data=json.dumps(new_profile), timeout=5)
+  assert r.status_code == 201, f"Expected status code 201, got {r.status_code}"
+  response = r.json()
+  assert "success" in response, f"Expected 'success' got {response}"
+
+  # Verify profile creation
+  r = requests.get(f"{API}/profiles", timeout=5)
+  assert r.status_code == 200, f"Expected status code 200, got {r.status_code}"
+  profiles = r.json()
+  created_profile = next(
+      (p for p in profiles if p["name"] == profile_name), None
+  )
+  assert created_profile is not None, f"Profile '{profile_name}' not found"
+
+#test for update profile when exists
+def test_update_profile(testrun, clean_profiles_dir): # pylint: disable=W0613
+  new_profile = load_json("new_profile.json")
+  updated_profile = load_json("updated_profile.json")
+  profile_name = new_profile["name"]
+  updated_profile_name = updated_profile["rename"]
+
+  #create the profile
+  r = requests.post(f"{API}/profiles", data=json.dumps(new_profile), timeout=5)
+  assert r.status_code == 201, f"Expected status code 201, got {r.status_code}"
+  response = r.json()
+  assert "success" in response, f"Expected 'success' got {response}"
+
+  assert profile_exists(profile_name), f"{profile_name} doesn't exist"
+
+  #update the profile
+  r = requests.post(
+      f"{API}/profiles",
+      data=json.dumps(updated_profile),
+      timeout=5)
+  assert r.status_code == 200, f"Expected status code 200, got {r.status_code}"
+  response = r.json()
+  assert "success" in response, f"Expected 'success' got {response}"
+
+  # verify profile update
+  r = requests.get(f"{API}/profiles", timeout=5)
+  assert r.status_code == 200, f"Expected status code 200, got {r.status_code}"
+  profiles = r.json()
+  updated_profile_check = next(
+    (p for p in profiles if p["name"] == updated_profile_name),
+    None
+  )
+  assert updated_profile_check is not None, f"Profile:{profile_name} not found"
+
+
+#test for delete profile
+def test_delete_profile(testrun, clean_profiles_dir): # pylint: disable=W0613
+#create the profile
+  new_profile = load_json("new_profile.json")
+  profile_name = new_profile["name"]
+  assert not profile_exists(profile_name), f"Profile '{profile_name}' exists"
+
+  r = requests.post(f"{API}/profiles", data=json.dumps(new_profile), timeout=5)
+  assert r.status_code == 201, f"Expected status code 201, got {r.status_code}"
+  response = r.json()
```

Comment:
Do not use the API to create the profile when testing the delete profile functionality. Maybe use a fixture to load profiles before starting testrun etc.